perf_test/sparse/KokkosSparse_gs.cpp

```diff
@@ -219,14 +219,22 @@ void runGS(const GS_Parameters& params) {
   KokkosSparse::Experimental::gauss_seidel_symbolic(
       &kh, nrows, nrows, A.graph.row_map, A.graph.entries,
       params.graph_symmetric);
-  double symbolicTime = timer.seconds();
-  std::cout << "\n*** Symbolic time: " << symbolicTime << '\n';
+  double symbolicLaunchTime = timer.seconds();
+  std::cout << "\n*** Symbolic launch time: " << symbolicLaunchTime << '\n';
+  timer.reset();
+  Kokkos::fence();
+  double symbolicComputeTime = timer.seconds();
+  std::cout << "\n*** Symbolic compute time: " << symbolicComputeTime << '\n';
```

Comment:
In general I would rewrite this as follows:
```
double symbolicLaunchTime = timer.seconds();
Kokkos::fence()l
double symbolicComputTime = timer.seconds() - symbolicLaunchTime;
std::cout << "\n*** Symbolic launch time: " << symbolicLaunchTime << '\n'
                << "\n*** Symbolic compute time: " << symbolicComputeTime << '\n';
```
It should be a bit more accurate since you do not really want to spend time printing and reseting the timer while the code executes in the GPU.

Reply:
Good catch. I'll update this in the stream support PR (#1906)

Reply:
With this change (https://github.com/kokkos/kokkos-kernels/pull/1906/commits/9a7775a2998cfeff07fc930a6481faca1d0eb19b) in 1906, we see:
```
$ ./perf_test/sparse/sparse_gs --cuda --amtx ~/matrix_market_files/pre2.mtx --point --sweeps 1

*** Symbolic launch time: 0.0344913

*** Symbolic compute time: 2.876e-06

*** Numeric launch time: 0.000436242

*** Numeric compute time: 0.000180281

*** Apply launch time: 0.00340607

*** Apply compute time: 3.2867e-05
Relative res norm: 1.42501e+38
```


```
$ ./perf_test/sparse/sparse_gs_lightweight --cuda --amtx ~/matrix_market_files/pre2.mtx --point --sweeps 1

*** Symbolic launch time: 0.034265

*** Symbolic compute time: 2.701e-06

*** Numeric launch time: 0.000432632

*** Numeric compute time: 0.000181905

*** Apply launch time: 0.000382276

*** Apply compute time: 0.00292852
Relative res norm: 3.06462e+38
```